src/main/java/core/basesyntax/service/DataConverterImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        return inputReport.stream()
+                .map(a -> a.trim())
+                .map(a -> a.split(","))
+                .filter(a -> isDataCorrect(a))

```

Comment:
The method 'isDataCorrect' returns a boolean but it is used in a filter operation, which may cause exceptions to be swallowed if data is incorrect. Filter should not throw exceptions.

Reply:
ok, done